src/generate/renders/tests/CliTestScenario.ts

```diff
@@ -253,6 +242,150 @@ export class CliTestScenario extends TemplateBase {
         }
     }
 
+    private GenerateTestResourceScenario(model: CodeModelAz, config: any, scenarioName: string) {
+        const exampleHandler = model.GetHandler().exampleHandler;
+        const commandParams = exampleHandler.GatherInternalResource();
+        config.unshift({ function: `setup_${scenarioName}` });
+        config.push({ function: `cleanup_${scenarioName}` });
+
+        const classInfo: string[] = [];
+        const initiates: string[] = [];
+
+        classInfo.push(`# Test class for ${scenarioName}`);
+        classInfo.push('@try_manual');
+        const testClassName = Capitalize(this.groupName) + scenarioName + 'Test';
+        classInfo.push('class ' + testClassName + '(ScenarioTest):');
+
+        const decorators: string[] = [];
+        const scenarioVariables = exampleHandler.GetResourcePool().getTestResourceVariables(config);
+        const kwargs = {};
+        for (const v of scenarioVariables) {
+            if (v === 'location') {
+                kwargs[v] = "'westus'";
```

Comment:
wonder why the location has to be westus ?

Reply:
this is an environment parameter for swagger test scenario and currently have no channel to know the right value. So here is just a guess strategy and assume the test scenario can be run in any location.
I think we can discuss with swagger team whether this can be added in test scenario in future oav release.

Reply:
@lirenhe any thoughts on this ?

Reply:
Similar condition in original solution using ResourceGroupPrepare. The location is implicitly westus since we don't assign it in codegen: https://github.com/Azure/azure-cli/blob/dev/src/azure-cli-testsdk/azure/cli/testsdk/preparers.py#L50

But I can add a config for this case if valuable.

Reply:
Added a parameter for customize location in test file